data/beaconrestapi/src/main/java/tech/pegasys/teku/beaconrestapi/handlers/v2/beacon/PostBlindedBlockV2.java

```diff
@@ -0,0 +1,151 @@
+      return;
+    }
+
+    final Optional<BroadcastValidationParameter> maybeBroadcastValidation =
+        request.getOptionalQueryParameter(PARAMETER_BROADCAST_VALIDATION);
+
+    final BroadcastValidationLevel broadcastValidationLevel =
+        maybeBroadcastValidation
+            .map(BroadcastValidationParameter::toInternal)
+            .orElse(BroadcastValidationLevel.GOSSIP);
+
+    final SignedBlockContainer requestBody = request.getRequestBody();
+
+    request.respondAsync(
+        validatorDataProvider
+            .submitSignedBlindedBlock(requestBody, broadcastValidationLevel)
+            .thenApply(
+                blockResult -> {
+                  if (blockResult.getRejectionReason().isEmpty()) {
+                    return AsyncApiResponse.respondWithCode(SC_OK);
+                  } else if (blockResult
+                      .getRejectionReason()
+                      .get()
+                      .equals(BlockImportResult.FailureReason.INTERNAL_ERROR.name())) {
+                    return AsyncApiResponse.respondWithError(
+                        SC_INTERNAL_SERVER_ERROR,
+                        "An internal error occurred, check the server logs for more details.");
+                  } else {
+                    return AsyncApiResponse.respondWithCode(SC_ACCEPTED);
+                  }
+                }));
+  }
+
+  private static EndpointMetadata getEndpointMetaData(
+      final Spec spec, final SchemaDefinitionCache schemaDefinitionCache) {
+    return EndpointMetadata.post(ROUTE)
+        .operationId("publishBlindedBlockV2")
+        .summary("Publish a signed blinded block")
+        .description(
+            """
+            Instructs the beacon node to use the components of the `SignedBlindedBeaconBlock` to construct and publish a \
+            `SignedBeaconBlock` by swapping out the `transactions_root` for the corresponding full list of `transactions`. \
+            The beacon node should broadcast a newly constructed `SignedBeaconBlock` to the beacon network, \
+            to be included in the beacon chain. The beacon node is not required to validate the signed \
+            `BeaconBlock`, and a successful response (20X) only indicates that the broadcast has been \
+            successful. The beacon node is expected to integrate the new block into its state, and \
+            therefore validate the block internally, however blocks which fail the validation are still \
+            broadcast but a different status code is returned (202). Pre-Bellatrix, this endpoint will accept \
+            a `SignedBeaconBlock`. The broadcast behaviour may be adjusted via the `broadcast_validation` \
+            query parameter.""")
+        .tags(TAG_VALIDATOR, TAG_VALIDATOR_REQUIRED)
+        .queryParam(PARAMETER_BROADCAST_VALIDATION)
+        .requestBodyType(
+            getSchemaDefinitionForAllSupportedMilestones(
+                schemaDefinitionCache,
+                "SignedBlindedBlock",
```

Comment:
SignedBlindedBeaconBlock?

Reply:
yes they are wrong also in V1... good catch